### ISSUE TITLE
Performance Problems

### ISSUE THREAD

**Issue by schayes04** [teambition/RRuleSwift#5]
Using this library to generate occurrences has generated significant performance problems for my app. It seems to get slowed down in the JS files and occasionally even crashes there.

**Comment by hongxinhope** [comment]
You're right, the poor performance is caused by the existence of JSContext, which is easy to cause a crash in widgets such as Today Extension. But for the host application, it's much better and
it performs well in most cases, it won't get crashes unless there is a problem elsewhere.

A good news is that we have developed a pure Swift algorithm to generate occurrences, and it can solve this problem. You can see it in [rrule.swift](https://github.com/sdq/rrule.swift), and we will 
integrate this two libraries together in the future if possible.
